pkg/southbound/gnmi/conn_manager.go

```diff
@@ -106,15 +103,21 @@ func (m *connManager) connect(ctx context.Context, target *topoapi.Object) (Conn
 	connID := newConnID()
 	log.Infof("Adding gNMI connection %s for the target %s", connID, target.ID)
 	conn := &conn{
-		id:     connID,
-		client: gnmiClient,
+		id:               connID,
+		client:           gnmiClient,
+		clientConn:       clientConn,
+		targetID:         target.ID,
+		connStateEventCh: make(chan connectivity.State),
 	}
 	m.connsMu.Lock()
 	m.conns[target.ID] = conn
-	m.clientConns[target.ID] = clientConn
 	m.connsMu.Unlock()
-	m.eventCh <- conn
+	m.eventCh <- ConnEvent{
+		Conn:      conn,
+		EventType: Connected,
+	}
 
+	go conn.processConnStateEvents()
```

Comment:
This should really be done in the `newConn` function since it’s an implementation detail of `conn`.

Reply:
I added a newConn function to return an instance of conn struct and inside that function, it starts this goroutine.